test/metabase/test.clj

```diff
@@ -287,6 +290,34 @@
   [clock & body]
   `(do-with-clock ~clock (fn [] ~@body)))
 
+(defn do-with-single-admin-user
+  [attributes thunk]
+  (let [existing-admin-memberships (db/select PermissionsGroupMembership :group_id (:id (perms-group/admin)))
+        _                          (db/simple-delete! PermissionsGroupMembership :group_id (:id (perms-group/admin)))
+        temp-admin                 (db/insert! User (merge (with-temp-defaults User)
+                                                           attributes
+                                                           {:is_superuser true}))
+        primary-key                (models/primary-key User)]
+    (try
+      (thunk temp-admin)
+      (finally
+        (db/delete! User primary-key (primary-key temp-admin))
+        (db/insert-many! PermissionsGroupMembership existing-admin-memberships)))))
```

Comment:
`insert-many!` does not call `post-insert` on inserted entities.

I guess in this particular case it's not important because `PermissionsGroupMembership/post-insert` do is making sure the `user.is_superuser=true`.

But maybe still consider doing multiple insert-one, or you can try using `metabase-enterprise.serialization.upsert/maybe-insert-many!` for this sort of things.

Reply:
My rationale with skipping the lifecycle hooks was that the `existing-admin-memberships` presumably would've been validated by `post-insert` and so forth when they were originally inserted. All we're doing here is restoring the DB state to what it was a few seconds previously.

Reply:
here's the commit https://github.com/metabase/metabase/pull/27524/commits/f012b9498bd67d4e062a7bafb68b33624807ed1a